src/lib/input/input.ts

```diff
@@ -232,7 +223,8 @@ export class MatInput implements MatFormFieldControl<any>, OnChanges, OnDestroy,
     const oldState = this.errorState;
     const ngControl = this.ngControl;
     const parent = this._parentFormGroup || this._parentForm;
-    const newState = ngControl && this.errorStateMatcher(ngControl.control as FormControl, parent);
+    const matcher = this.errorStateMatcher || this._defaultErrorStateMatcher;
+    const newState = ngControl && matcher.isErrorState(ngControl, parent);
```

Comment:
`isErrorState` accepts null `ngControl`, so no need to check here

Reply:
Done.